cornflow-dags/DAG/rostering/core/instance.py

```diff
@@ -74,47 +75,55 @@ def from_dict(cls, data: dict) -> "Instance":
 
         data_p["parameters"] = pickle.loads(pickle.dumps(data["parameters"], -1))
 
-        if data.get("skill_demand"):
+        if data.get("skill_demand") and data_p["requirements"]["skill_demand"] == True:
```

Comment:
`if data.get("skill_demand") and data_p["requirements"]["skill_demand"] == True:` can simply be written
`if data.get("skill_demand") and data_p["requirements"]["skill_demand"]:`.
Also, I have no doubt that this works but maybe it would be cleaner to activate/deactivate the constraints when constructing the model instead of emptying the tables when the parameters is False ?